webapi/src/main/scala/org/knora/webapi/responders/v1/ValueUtilV1.scala

```diff
@@ -120,6 +120,24 @@ class ValueUtilV1(private val settings: SettingsImpl) {
     }
 
     /**
+      * Creates a URL pointing to the given resource class icon. From the resource class Iri it gets the ontology specific path, i.e. the ontology name.
+      * If the resource class Iri is "http://www.knora.org/ontology/knora-base#Region", the ontology name would be "knora-base".
+      * To the base path, the icon name is appended. In case of a region with the icon name "region.gif",
+      * "http://salsahapp:port/project-icons-basepath/knora-base/region.gif" is returned.
+      *
+      * @param resourceClassIri the Iri of the resource class in question.
+      * @param iconsSrc the name of the icon file.
+      */
+    def makeResourceClassIconURL(resourceClassIri: IRI, iconsSrc: String): IRI = {
+        // get ontology name, e.g. "knora-base" from "http://www.knora.org/ontology/knora-base#Region"
+        // add +1 to ignore the slash
+        val ontologyName = resourceClassIri.substring(resourceClassIri.lastIndexOf('/') + 1, resourceClassIri.lastIndexOf('#'))
```

Comment:
This won't work for all IRIs. The last path component of an IRI doesn't have to be a unique name for the ontology. For example, the IRI of `foaf:Person` is `http://xmlns.com/foaf/0.1/Person`; the last path component is `0.1`, which isn't the name of the ontology. Either we need to impose this as a requirement on all ontologies used with Knora (in which case it needs to be documented), or we need to find another way to associate resource classes with ontology names.